src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -388,48 +388,100 @@ public ContentItemDisplay GetEmptyByKey(Guid contentTypeKey, int parentId)
+            var mapped = MapToDisplay(emptyContent);
+            return CleanContentItemDisplay(mapped);
         }
 
         /// <summary>
-        /// Gets a collection of empty content items for all document types.
+        /// Gets an empty <see cref="ContentItemDisplay"/> for each content type in the IEnumerable, all with the same parent ID
         /// </summary>
-        /// <param name="contentTypeKeys"></param>
+        /// <remarks>Will attempt to re-use the same permissions for every content as long as the path and user are the same</remarks>
+        /// <param name="contentTypes"></param>
         /// <param name="parentId"></param>
-        [OutgoingEditorModelEvent]
-        public IDictionary<Guid, ContentItemDisplay> GetEmptyByKeys([FromUri] Guid[] contentTypeKeys, [FromUri] int parentId)
+        /// <returns></returns>
+        private IEnumerable<ContentItemDisplay> GetEmpties(IEnumerable<IContentType> contentTypes, int parentId)
         {
-            var result = new Dictionary<Guid, ContentItemDisplay>();
-
-            using var scope = _scopeProvider.CreateScope(autoComplete: true);
-            var contentTypes = Services.ContentTypeService.GetAll(contentTypeKeys).ToList();
+            var result = new List<ContentItemDisplay>();
+            var userId = Security.GetUserId().ResultOr(0);
+            var currentUser = Security.CurrentUser;
+            // We know that if the ID is less than 0 the parent is null.
+            // Since this is called with parent ID it's safe to assume that the parent is the same for all the content types.
+            var parent = parentId > 0 ? Services.ContentService.GetById(parentId) : null;
+            EntityPermissionSet permissions = null;
 
             foreach (var contentType in contentTypes)
             {
-                if (contentType is null)
+                var emptyContent = Services.ContentService.Create("", parentId, contentType, userId);
+
+                if (emptyContent is null)
                 {
                     throw new HttpResponseException(HttpStatusCode.NotFound);
                 }
 
-                result.Add(contentType.Key, GetEmpty(contentType, parentId));
+                // Get the path for the content, we get this for each just in case some content has Identity and a different path
+                // Since if that's the case we might not be able to re-use the permissions.
+                string path;
+                if (emptyContent.HasIdentity)
+                {
+                    path = emptyContent.Path;
+                }
+                else
+                {
+                    path = parent == null ? "-1" : parent.Path;
+                }
+
+                // Only assign permissions once, the idea is that we'll be able to re-use the same permissions most of the time.
+                if (permissions is null)
+                {
+                    permissions = Services.UserService.GetPermissionsForPath(currentUser, path);
```

Comment:
Would it be simpler if permissions was a `Dictionary<string, EntityPermissionCollection>` and you would just lookup permissions for any unique path?

Reply:
Hmm I don't think it would necessarily be simpler, looking at it now, the path used to get the permissions is the parent path, and since the parent here is always the same, we actually don't need to keep creating the path, so we don't need to check if permission is null, we can just get the path and permissions once before the loop and use that for all of them.

However, I think using the `Dictionary<string, EntityPermissionCollection>` might be a good idea, mainly because it would make it easier to make optimisations in future if, for instance, we wanted to map multiple IContent where some have a different path then the dictionary approach would allow for that, whereas the current approach really only allows for a single path.